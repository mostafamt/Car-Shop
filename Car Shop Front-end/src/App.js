import React from "react";
import ButtonAppBar from "./ButtonAppBar";
import Login from "./components/Login";
import { AuthProvider } from "./context/AuthContext";

function App() {
  return (
    <AuthProvider>
      <ButtonAppBar />
      <Login />
    </AuthProvider>
  );
}

export default App;
