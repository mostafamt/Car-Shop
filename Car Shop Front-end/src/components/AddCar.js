import React from "react";

import Button from "@mui/material/Button";
import AddIcon from "@mui/icons-material/Add";

import { useFormik } from "formik";
import Dialog from "./Dialog";

const AddCar = (props) => {
  const [open, setOpen] = React.useState(false);

  const handleClose = () => {
    setOpen(false);
  };

  const formik = useFormik({
    initialValues: {
      brand: "",
      model: "",
      color: "",
      productionYear: "",
      price: "",
    },
    onSubmit: (values) => {
      props.addCar(values);
      handleClose();
      formik.handleReset();
    },
  });

  return (
    <div>
      <Button
        variant="contained"
        onClick={() => setOpen(true)}
        startIcon={<AddIcon />}
      >
        New Car
      </Button>
      <Dialog open={open} onClose={handleClose} formik={formik} />
    </div>
  );
};

export default AddCar;
