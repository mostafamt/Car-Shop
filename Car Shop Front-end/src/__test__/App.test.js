import { fireEvent, render, screen } from "@testing-library/react";
import App from "../App";

import TestRenderer from "react-test-renderer";
import AddCar from "../components/AddCar";

test("open add car modal form", () => {
  render(<App />);
  fireEvent.click(screen.getByText("New Car"));
  expect(screen.getByRole("dialog")).toHaveTextContent("New Car");
});

test("renders a snapshot", () => {
  const tree = TestRenderer.create(<AddCar />).toJSON();
  expect(tree).toMatchSnapshot();
});
