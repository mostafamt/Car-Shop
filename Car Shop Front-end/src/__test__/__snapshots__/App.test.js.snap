// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="AddIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
    </span>
    New Car
  </button>
</div>
`;
